db/dataset.py

```diff
@@ -183,7 +183,15 @@ def get_public_datasets():
             row = dict(row)
             dataset_id = row["id"].encode('utf-8')
             row["status"] = dataset_status[dataset_id]
```

Comment:
i've been playing around with this a bit. Here's a query which gets status as well, using a `LATERAL JOIN`, showing a null status if the dataset hasn't been submitted

```sql
explain analyze SELECT dataset.id :: TEXT
     , dataset.name
     , dataset.description
     , dataset.author
     , "user".musicbrainz_id AS author_name
     , dataset.created
     , job.status
  FROM dataset
  JOIN "user"
    ON "user".id = dataset.author
  LEFT JOIN LATERAL (SELECT status
                       FROM dataset_eval_jobs
                      WHERE dataset.id = dataset_eval_jobs.dataset_id
                   ORDER BY updated DESC
                      LIMIT 1)
                   AS job ON TRUE
 WHERE dataset.public = 't'
  and job.status = 'pending'
ORDER BY dataset.created DESC
OFFSET 0 LIMIT 10
```

You can modify the `WHERE job.status` and `OFFSET` and `LIMIT` to choose what items you show